src/components/Calendar/hooks/useCalendarState.ts

```diff
@@ -172,10 +207,10 @@ export function useCalendarState(props: CalendarStateOptions): CalendarState {
             return this.isInvalid(next);
         },
         isSelected(date: DateTime) {
-            return (
-                Boolean(value) &&
-                date.isSame(value ?? undefined, currentMode) &&
-                !this.isCellDisabled(date)
+            return Boolean(
+                value &&
+                    date.isSame(value.timeZone(timeZone), currentMode) &&
```

Comment:
[isSame](https://github.com/gravity-ui/date-utils/blob/986b63f1996a86a57ce261739a27eb05a61ae2c4/src/dateTime/dateTime.ts#L234) does not depend on the time zone